text/0126-Configurable-Metric-Aggregations.md

```diff
@@ -0,0 +1,99 @@
+# A Proposal For SDK Support for Configurable Batching and Aggregations (Basic Views)
+
+Add support to the default SDK for the ability to configure Metric Aggregations.
+
+## Motivation
```

Comment:
I can see also another config, simpler:
* Change default temporality for instrument type - separated than view or selector.
* Enable/Disable an instrument default aggregation.

Maybe this are out of scope, but kind of think that they may be configured separately than the views.

Reply:
I was thinking that the "disable" use-case could just be done by setting the aggregation to a no-op aggregator.

Reply:
Worried that having this "only one" view requirement and relying on that to enable/disable default aggregations, or even doing some merging of "Views" may be too tricky and too restrictive for the future.

I would like you to think how would you do this if full views API is supported.

Bogdan self brainstorming:

Maybe having a way to retrieve all the configured views (InstrumentDescriptor + View) and allow:
* "Delete" - the user will use this to disable default aggregation for example
* You can clone/change the retrieved "View" after deleting it and set it back to update the "temporality"

It seems a lot of work to just disable "default aggregations". Maybe treat "default views" a bit special, and offer specific APIs to manipulate "all of them" like:
* "default views temporality" - these APIs may include selectors to facilitate.
* "disable all default views"
But this way we make it clear that these "DefaultViewsConfiguration" API refers to only the default views and you no longer need the "one view only" requirement.

Reply:
Another way is to have a "strategy" param to the API: replace all (only supported option now).

Reply:
I like the idea of considering this OTEP the "default views configuration". Maybe we can just simplify it down to just selecting by instrument type for this?

Reply:
I think the selector can be fully defined, the API should include in the name that this applies only to default views, also operation that we accept:
* Update temporality for selected instruments
* Disable/Delete default views for selected instruments.

Changing labels is a complex operation that we should probably not support on the default views, recommended way should be deleted the default view and add a new view.

Reply:
I can be convinced that labels can be also an operation available for default views, but probably we can discuss that later. Let's start with something then incremently add more stuff

Reply:
Also if for selector only type is important now, we can start only with that. But these decisions I think can be made in the specs

Reply:
Agree with leaving details to the specs. I'd be happy to drop label support for now, to keep things ultra-simple and build from there. Do you want this OTEP updated to reflect that, or is that something we can wrangle out in the specs?

Reply:
Just a sentence that say details still to be polished in specs is more than enough

Reply:
I added a few more open questions to capture the discussion, and a note that we will resolve them in the spec. Sound ok?